kong/plugins/opentelemetry/handler.lua

```diff
@@ -0,0 +1,188 @@
+local new_tab = require "table.new"
+local http = require "resty.http"
+local clone = require "table.clone"
+local otlp = require "kong.plugins.opentelemetry.otlp"
+local propagation = require "kong.tracing.propagation"
+local tablepool = require "tablepool"
+
+local ngx = ngx
+local kong = kong
+local ngx_log = ngx.log
+local ngx_ERR = ngx.ERR
+local ngx_DEBUG = ngx.DEBUG
+local ngx_now = ngx.now
+local ngx_update_time = ngx.update_time
+local ngx_req = ngx.req
+local ngx_get_headers = ngx_req.get_headers
+local timer_at = ngx.timer.at
+local clear = table.clear
+local propagation_parse = propagation.parse
+local propagation_set = propagation.set
+local tablepool_release = tablepool.release
+local tablepool_fetch = tablepool.fetch
+local null = ngx.null
+local encode_traces = otlp.encode_traces
+local transform_span = otlp.transform_span
+
+local POOL_BATCH_SPANS = "KONG_OTLP_BATCH_SPANS"
+local _log_prefix = "[otel] "
+
+local OpenTelemetryHandler = {
+  VERSION = "0.1.0",
+  PRIORITY = 14,
+}
+
+local default_headers = {
+  ["Content-Type"] = "application/x-protobuf",
+}
+
+-- worker-level spans cache
+local spans_cache = new_tab(5000, 0)
```

Comment:
This table is more like a buffer than `cache`.
A better name would be `not_synced_spans`, or just `spans_buffer`.

Reply:
spans_buffer sounds better